urllib3/util/ssl_.py

```diff
@@ -17,6 +18,23 @@
 
 import errno
 import warnings
+import hmac
+
+if hasattr(hmac, 'compare_digest'):
+    _const_compare_digest = hmac.compare_digest
+else:
+    def _const_compare_digest(a, b):
+        """
+        Compare two digests of equal length in constant time.
+
+        Returns True if the digests match, and False otherwise.
+        """
+        result = abs(len(a) - len(b))
+        for l, r in six.moves.zip(bytearray(a), bytearray(b)):
```

Comment:
Reason why having superfluous logic is bad: Something someday will rely on that undocumented/unexpected quirk, and when we refactor and get rid of it, things will break. (e.g. length checks)

Reason why using six unnecessarily is bad: Someday we're going to remove our usage of six, and fewer usages to remove is better.

This entire PR is an exercise in doing things that are not necessary, so allocating an extra tuple is not exactly contradictory with the spirit. :)